Subject: [PATCH] trata de borrar doble llamado a alltasks() por la recomposicion de TaskList pero falla con 401. No se por que
---
Index: frontend/viewmodel/task/impl/src/commonMain/kotlin/TaskViewModelImpl.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/viewmodel/task/impl/src/commonMain/kotlin/TaskViewModelImpl.kt b/frontend/viewmodel/task/impl/src/commonMain/kotlin/TaskViewModelImpl.kt
--- a/frontend/viewmodel/task/impl/src/commonMain/kotlin/TaskViewModelImpl.kt	(revision a5754f4673d99d7fa87b628b9815a0ae956052e9)
+++ b/frontend/viewmodel/task/impl/src/commonMain/kotlin/TaskViewModelImpl.kt	(date 1746651586822)
@@ -1,11 +1,32 @@
 package hernanbosqued.frontend.viewmodel.task.impl
 
 import hernanbosqued.domain.IdTask
+import hernanbosqued.domain.UserData
 import hernanbosqued.frontend.viewmodel.task.TaskUseCase
 import hernanbosqued.frontend.viewmodel.task.TaskViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
 
 class TaskViewModelImpl(
     private val taskUseCase: TaskUseCase,
+    coroutineScope: CoroutineScope
 ) : TaskViewModel {
-    override suspend fun getTasks(): List<IdTask> = taskUseCase.allTasks()
+    private val _tasks = MutableStateFlow<List<IdTask>>(emptyList())
+    override val tasks: StateFlow<List<IdTask>> = _tasks.asStateFlow()
+
+    init {
+        println("=========TASKVIEWMODEL INIT================")
+        coroutineScope.launch {
+            _tasks.value = taskUseCase.allTasks()
+        }
+
+//        taskUseCase.userData
+//            .onEach { _authState.value = it }
+//            .launchIn(coroutineScope)
+    }
 }
Index: frontend/viewmodel/task/public/src/commonMain/kotlin/TaskViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/viewmodel/task/public/src/commonMain/kotlin/TaskViewModel.kt b/frontend/viewmodel/task/public/src/commonMain/kotlin/TaskViewModel.kt
--- a/frontend/viewmodel/task/public/src/commonMain/kotlin/TaskViewModel.kt	(revision a5754f4673d99d7fa87b628b9815a0ae956052e9)
+++ b/frontend/viewmodel/task/public/src/commonMain/kotlin/TaskViewModel.kt	(date 1746651118859)
@@ -1,7 +1,8 @@
 package hernanbosqued.frontend.viewmodel.task
 
 import hernanbosqued.domain.IdTask
+import kotlinx.coroutines.flow.StateFlow
 
 interface TaskViewModel {
-    suspend fun getTasks(): List<IdTask>
+    val tasks: StateFlow<List<IdTask>>
 }
Index: frontend/viewmodel/task/impl/build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/viewmodel/task/impl/build.gradle.kts b/frontend/viewmodel/task/impl/build.gradle.kts
--- a/frontend/viewmodel/task/impl/build.gradle.kts	(revision a5754f4673d99d7fa87b628b9815a0ae956052e9)
+++ b/frontend/viewmodel/task/impl/build.gradle.kts	(date 1746650682290)
@@ -17,6 +17,8 @@
 
     sourceSets {
         commonMain.dependencies {
+            implementation(libs.kotlinx.coroutines.core)
+
             implementation(project(":domain"))
             implementation(project(":frontend:viewmodel:task:public"))
             implementation(project(":frontend:use_case:task:public"))
Index: frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/TaskList.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/TaskList.kt b/frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/TaskList.kt
--- a/frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/TaskList.kt	(revision a5754f4673d99d7fa87b628b9815a0ae956052e9)
+++ b/frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/TaskList.kt	(date 1746651118849)
@@ -9,6 +9,7 @@
 import androidx.compose.foundation.rememberScrollbarAdapter
 import androidx.compose.foundation.verticalScroll
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
@@ -23,16 +24,10 @@
 import org.koin.compose.koinInject
 
 @Composable
-@Preview
 fun TaskList() {
     val viewModel = koinInject<TaskViewModel>()
+    val tasks by viewModel.tasks.collectAsState()
     val scrollState = rememberScrollState()
-    var tasks by remember { mutableStateOf<List<IdTask>>(emptyList()) }
-    val coroutineScope = rememberCoroutineScope()
-
-    coroutineScope.launch {
-        tasks = viewModel.getTasks()
-    }
 
     Box(
         modifier = Modifier.Companion.fillMaxSize(),
Index: frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/App.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/App.kt b/frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/App.kt
--- a/frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/App.kt	(revision a5754f4673d99d7fa87b628b9815a0ae956052e9)
+++ b/frontend/ui/src/commonMain/kotlin/hernanbosqued/frontend/ui/App.kt	(date 1746650006157)
@@ -12,13 +12,17 @@
 @Composable
 fun App() {
     KoinContext {
-        val authViewModel = koinInject<AuthViewModel>()
-        val authState by authViewModel.authState.collectAsState()
-
         MaterialTheme {
             Column {
+                val authViewModel = koinInject<AuthViewModel>()
+                val authState by authViewModel.authState.collectAsState()
+
+                println("---------------AUTHSTATE----------$authState")
+
                 AuthRow()
                 if (authState != null) {
+                    println("---------------SE COMPONE TASKLIST----------")
+
                     TaskList()
                 }
             }
Index: frontend/viewmodel/task/public/build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/viewmodel/task/public/build.gradle.kts b/frontend/viewmodel/task/public/build.gradle.kts
--- a/frontend/viewmodel/task/public/build.gradle.kts	(revision a5754f4673d99d7fa87b628b9815a0ae956052e9)
+++ b/frontend/viewmodel/task/public/build.gradle.kts	(date 1746651118854)
@@ -17,11 +17,9 @@
 
     sourceSets {
         commonMain.dependencies {
+            implementation(libs.kotlinx.coroutines.core)
+
             implementation(project(":domain"))
         }
-
-        wasmJsMain.dependencies {
-            implementation(libs.kotlinx.browser)
-        }
     }
 }
Index: frontend/viewmodel/task/di/src/commonMain/kotlin/TaskViewModelModule.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/viewmodel/task/di/src/commonMain/kotlin/TaskViewModelModule.kt b/frontend/viewmodel/task/di/src/commonMain/kotlin/TaskViewModelModule.kt
--- a/frontend/viewmodel/task/di/src/commonMain/kotlin/TaskViewModelModule.kt	(revision a5754f4673d99d7fa87b628b9815a0ae956052e9)
+++ b/frontend/viewmodel/task/di/src/commonMain/kotlin/TaskViewModelModule.kt	(date 1746651178387)
@@ -9,7 +9,10 @@
         module {
             single<TaskViewModel> {
                 println("Atlanta 2")
-                TaskViewModelImpl(get())
+                TaskViewModelImpl(
+                    taskUseCase = get(),
+                    coroutineScope = get()
+                )
             }
         }
 }
Index: frontend/viewmodel/task/di/build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/viewmodel/task/di/build.gradle.kts b/frontend/viewmodel/task/di/build.gradle.kts
--- a/frontend/viewmodel/task/di/build.gradle.kts	(revision a5754f4673d99d7fa87b628b9815a0ae956052e9)
+++ b/frontend/viewmodel/task/di/build.gradle.kts	(date 1746651155523)
@@ -18,15 +18,11 @@
     sourceSets {
         commonMain.dependencies {
             implementation(libs.koin.core)
+            implementation(libs.kotlinx.coroutines.core)
+
             api(project(":frontend:viewmodel:task:public"))
             implementation(project(":frontend:viewmodel:task:impl"))
             implementation(project(":frontend:use_case:task:public"))
         }
-
-        jvmMain.dependencies {
-        }
-
-        wasmJsMain.dependencies {
-        }
     }
 }
